### PR TITLE
Workorders

### PR DESCRIPTION
**Pull request by Dartellum**
Bone working now with new logic. Added stamp to carve and made changes for bone to work. Changed qun in base-crafting.yaml to pollen to make it work for remedy. Modified my yaml.

### CODE REVIEW

File: carve.lic [hunk 1]
```diff
@@ -50,27 +60,31 @@ class Carve
...
+    echo("Type is #{@type} and tool is #{@tool}") if @debug_mode
+    carve("cut my #{@type} with my #{@tool}")
   end
 
   def carve(command)
     waitrt?
     case bput(command,
-              'rough, jagged edges',
-              'determine it is no longer level',
+#             'rough, jagged (edges|shards)',
```
Comment (rpherbig) [inline carve.lic]: Can you remove the commented-out code?

File: carve.lic [hunk 2]
```diff
@@ -12,12 +12,22 @@ class Carve
     settings = get_settings
     @bag = settings.crafting_container
     @belt = settings.engineering_belt
+    @use_bone = settings.use_bone
```
Comment (rpherbig) [inline carve.lic]: Given how many settings we have, I'm not sure `use_bone` is descriptive/specific enough. What about `use_bone_for_carving`?

File: workorders.lic [hunk 3]
```diff
@@ -158,10 +171,17 @@ class WorkOrders
   end
 
   def find_recipe(info, item, quantity)
-    recipe = @recipes.find { |r| r['name'] == item }
-    items_per_stock = info['stock-volume'] / recipe['volume']
-    spare_stock = (info['stock-volume'] % recipe['volume']).nonzero?
-    scrap = spare_stock || (quantity % items_per_stock).nonzero?
+    if !@use_bone
```
Comment (rpherbig) [inline workorders.lic]: Is the only difference in these two cases `stock-volume` vs `stock-bone-volume`?

File: workorders.lic [hunk 4]
```diff
@@ -158,10 +171,17 @@ class WorkOrders
...
+      recipe = @recipes.find { |r| r['name'] == item }
+      items_per_stock = info['stock-volume'] / recipe['volume']
+      spare_stock = (info['stock-volume'] % recipe['volume']).nonzero?
```
Comment (rpherbig) [inline workorders.lic]: What do you think about renaming `stock-volume` to `stock-rock-volume`?

**Comment by Dartellum** [comment]
Sure, renaming now.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Comment by Dartellum** [comment]
Debug_mode is not set. It is not displaying the echos with this command 
either. I can remove the echos or set debug_mode.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Comment by Dartellum** [comment]
I do not have that on my side.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Comment by Dartellum** [comment]
Let's come back to this after the other pieces are corrected. Yes, the 
only difference is the stock piece.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

File: forge.lic [hunk 5]
```diff
@@ -141,10 +148,17 @@ class Forge
...
-      waitrt?
-      stow_item('shovel')
+      get_item(@tool)
+        if @tool == 'tong'
+          bput("adjust #{@tool}","You lock")
+          bput("push fuel with my #{@tool}", "Roundtime")
+          waitrt?
+          bput("adjust #{@tool}","With a yank")
+        else
+          bput("push fuel with my #{@tong}", "Roundtime")
```
Comment (rpherbig) [inline forge.lic]: Should this (and line 160) be `@tool`?

File: forge.lic [hunk 6]
```diff
@@ -45,8 +45,15 @@ class Forge
     @bag = settings.crafting_container
     @belt = settings.forging_belt
     @hammer = settings.forging_tools.find { |item| /hammer/ =~ item }
+    @tong_shovel = settings.use_adjustable_tong
 
     @item = args.noun
+    
+    if @tong_shovel
+      @tool = 'tong'
```
Comment (rpherbig) [inline forge.lic]: I see quite a few other places that 'tongs' are used. Should they also reference `@tool`?

**Comment by Dartellum** [comment]
I corrected.  In this example referenced, @tong_shovel is the setting in 
the yaml, true or false with no setting false. @tool is then set to tong 
or shovel based on that. Further done I did have @tong in. I started 
with @tong and decided it did not make sense to use since it could also 
have the value of shovel.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

File: smelt.lic [hunk 7]
```diff
@@ -91,10 +101,23 @@ class Smelt
   def fuel
     pause 1
     waitrt?
-    get_crafting_item('shovel', @belt)
-    bput('push fuel with my shovel', 'roundtime')
+    if @tool == 'tong'
+      get_crafting_item(@tool, @belt)
+      bput("adjust my #{@tool}","You lock")
+    else
+      stow_crafting_item(@tool, @bag, @belt)
```
Comment (rpherbig) [inline smelt.lic]: Why stow `@tool` only to get it again immediately?

The same question applies to lines 118 and 119.

**Comment by Dartellum** [comment]
Yes!

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Comment by Dartellum** [comment]
You either get the tong and adjust, push fuel with tong, then adjust 
again or get the shovel and push fuel normally.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Comment by Dartellum** [comment]
Have you added any of the changes yet? I want to do a pull but not if the files were not merged/changed. You can piece the code in if you do not want to apply whole scale. I am fine with that.

**Comment by rcuhljr** [comment]
As long as the PR is open we haven't pulled in the code from it.

**Comment by rcuhljr** [comment]
This guys been open for awhile, if you want I can tackle just getting it merged and fixing the issues when I get time. I can probably do it tomorrow.

**Comment by Dartellum** [comment]
I did not do this section, however, I will correct and re-push.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Comment by Dartellum** [comment]
You are right. I just ran through again and noticed it did not do what I 
wanted it do.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Comment by Dartellum** [comment]
My last update I made qun pollen and ithor root, four recipes, false for work_order since they need additional processing with Society tools or pyramid.

**Comment by rpherbig** [comment]
@Dartellum Can this PR be closed in favor of #1008 ?

**Comment by Dartellum** [comment]
Yes.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

**Pull request closed without merging**